src/docs/development/platform-integration/c-interop.md

```diff
@@ -351,6 +351,35 @@ in binary form, use the following instructions:
 
 This feature is not yet supported for web plugins.
 
+## FAQ
+
+### Android APK size (shared object compression)
+
+Android guidelines in general recommend distributing native shared objects uncompressed because that actually saves on device space. Shared objects can be directly loaded from the APK instead of unpacking them on device into a temporary location and then loading. APKs are additionally packed in transit - that is why you should be looking at download size. 
+
+Flutter APKs by default don't follow these guidelines (we are in discussions about that) and compress `libflutter.so` and `libapp.so` - this leads to smaller APK size but larger on device size.
+
+Using `dart:ffi` inverts this default setting, and stops the compression of `libflutter.so`, `libapp.so`, and any user-added shared objects. To re-enable compression change `android/app/src/main/AndroidManifest.xml` in the following way.
```

Comment:
I would like to understand how `dart:ffi` impact this setting. Right now it is completely magical that this happens.

Reply:
Without `tools:replace="android:extractNativeLibs"` we get a source:

> is also present at [io.objectbox:objectbox-android:2.5.0] AndroidManifest.xml:22:9-42 value=(false).

@greenrobot what is `objectbox-android`? And where do I find its source code?

Reply:
> This is because ObjectBox adds `extractNativeLibs="false"` to your `AndroidManifest.xml` as recommended by Google.

https://docs.objectbox.io/faq#how-much-does-objectbox-add-to-my-apk-size

This might very well be the cause.

Reply:
Based on this understanding I think we should rewrite this text and remove paragraphs that blame `dart:ffi`. Instead we can say that some libraries do this and it can be counteracted by doing `extractNativeLibs="true"`